pyproject.toml

```
@@ -77,6 +77,7 @@ h5netcdf = ">=0.11"
 dask = ">=2022.1"
 pyproj = ">=3.5.0"
 bump-my-version = ">=0.10.0"
+vcrpy = "^6.0.1"
```

@jhkennedy, is this another case where we want to add the dep to `ci/enviroment-dev.yaml` too? How about the version pin?

---

yes, same applies -- we should add it to the `ci/environment-*` YAMLs and should remove the upper bound
```suggestion
vcrpy = ">=6.0.1"
```

---

This might be a topic for a separate discussion, but I generally advise against `>=`-only constraints because that can open us up to unwittingly introducing breaking changes by pulling in the next major release of a dependency. I generally recommend `~=`, but without the patch version included. For example, in this particular case, `~=6.0`, which is equivalent to `>=6.0,<7`.

---

@chuckwondo the poetry `^` is fine and roughly equivalent to `~=` though behaves better with patch-level pins. 

The separate discussion is here: https://github.com/nsidc/earthaccess/pull/468

and generally I agree with you for *applications* (although there `==` with dependabot or similar is better) but not *libraries* like Earthaccess. With an upper bound, we *guarantee* Earthaccess will break in the future while a major release *may or may not* cause Earthaccess to break and is better caught with regular testing